lib/internal/test_runner/reporter/spec.js

```diff
@@ -36,27 +37,25 @@ const symbols = {
   'arrow:right': '\u25B6 ',
   'hyphen:minus': '\uFE63 ',
 };
-class SpecReporter extends Transform {
-  #stack = [];
-  #reported = [];
-  #indentMemo = new SafeMap();
-  #failedTests = [];
-  #cwd = process.cwd();
-
-  constructor() {
-    super({ __proto__: null, writableObjectMode: true });
-  }
 
-  #indent(nesting) {
-    let value = this.#indentMemo.get(nesting);
+module.exports = async function* specReporter(source) {
+  let stack = [];
+  let reported = [];
+  let indentMemo = new SafeMap();
+  let failedTests = [];
+  let cwd = process.cwd();
+
+  function indent(nesting) {
```

Comment:
this means all of these functions are defined anew every time this reporter is called, instead of just once at module startup.

Reply:
these functions require closure variables

Reply:
then if you're switching to a non-class-based model, the functions should take everything as arguments, so they can be defined at module level.